src/cbmc/cbmc_parse_options.cpp

```diff
@@ -553,6 +550,34 @@ int cbmc_parse_optionst::doit()
   if(set_properties())
     return CPROVER_EXIT_SET_PROPERTIES_FAILED;
 
+  // At this point, our goto-model should be in core-goto form (all of the
+  // transformations have been run and the program is ready to be given to the
+  // solver).
+  if(options.is_set("export-core-goto"))
+  {
+    auto core_goto_filename = options.get_option("export-core-goto");
+    if(core_goto_filename.empty())
+    {
+      log.error()
+        << "ERROR: Please provide a filename to write the goto-binary to."
+        << messaget::eom;
+      return CPROVER_EXIT_INTERNAL_ERROR;
+    }
+
+    auto success =
+      !write_goto_binary(core_goto_filename, goto_model, ui_message_handler);
+
+    if(!success)
+    {
+      log.error() << "ERROR: Unable to export goto-program in file "
+                  << core_goto_filename << messaget::eom;
+      return CPROVER_EXIT_INTERNAL_ERROR;
+    }
+    log.status() << "Exported goto-program in core-goto form at "
```

Comment:
:pick: I'd put these 2 lines in an else block. Just so that the two blocks of code of equivalent control flow significance are at the same level of indentation.